src/main.rs

```diff
@@ -81,15 +81,17 @@ fn main() {
     setup_logger(opt.loglevel);
     info!("[main] shimmy says hi!");
 
-    match fork() {
-        Ok(ForkResult::Parent { child }) => {
-            // Main process (cont.)
-            write_runtime_pidfile(opt.pidfile, child);
-            exit(0);
-        }
-        Ok(ForkResult::Child) => (), // Shim process
-        Err(err) => panic!("fork() of the shim process failed: {}", err),
-    };
+    unsafe {
```

Comment:
Could be simplified to `match unsafe { fork() } { ... }`. I was upgrading a similar code recently [here](https://github.com/iximiuz/reapme/commit/84bcb62f4d8d8ccbe0688377b9d770de9c4983f1).